staging/src/k8s.io/apimachinery/pkg/labels/selector_test.go

```diff
@@ -573,3 +573,39 @@ func TestAdd(t *testing.T) {
 		}
 	}
 }
+
+func TestSafeSort(t *testing.T) {
+	tests := []struct {
+		name string
+		in   []string
+		want []string
+	}{
+		{
+			name: "nil strings",
+			in:   nil,
+			want: nil,
+		},
+		{
+			name: "ordered strings",
+			in:   []string{"bar", "foo"},
+			want: []string{"bar", "foo"},
+		},
+		{
+			name: "unordered strings",
+			in:   []string{"foo", "bar"},
+			want: []string{"bar", "foo"},
+		},
+		{
+			name: "duplicated strings",
+			in:   []string{"foo", "bar", "foo", "bar"},
+			want: []string{"bar", "bar", "foo", "foo"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := safeSort(tt.in); !reflect.DeepEqual(got, tt.want) {
```

Comment:
This test doesn't verify that tt.in is unmodified?

Reply:
good point. addressed.